lib/broadway_kafka/brod_client.ex

```diff
@@ -169,6 +169,13 @@ defmodule BroadwayKafka.BrodClient do
 
   defp lookup_offset(hosts, topic, partition, policy, client_config) do
     case :brod.resolve_offset(hosts, topic, partition, policy, client_config) do
+      {:ok, -1} ->
+        # `:brod.resolve_offset` returns -1 when asked to resolve a timestamp newer
+        # than all the messages in the partition.
+        # -1 is not a valid offset you can use with `:brod.fetch` so we need to
+        # resolve the latest offset instead
+        lookup_offset(hosts, topic, partition, :latest, client_config)
```

Comment:
Would anyone want to reset to earliest instead? Maybe it should be `{:timestamp, timestamp, :earliest | :latest}`?

Reply:
Following the docs here https://github.com/kafka4beam/brod/blob/master/src/brod.erl#L1125-L1157

> The function returns the offset of the first message
with the given timestamp, or of the first message after
the given timestamp

I assumed the logical conclusion was to go for the latest, because it's the one after the timestamp you asked for

Happy to add the option if you think it's best

Reply:
Makes sense. Fine by me as is then. :)

Reply:
Awesome ❤️ , let me know if anything else needs to be done before this can be merged